great_expectations/core/factory/checkpoint_factory.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from great_expectations._docs_decorators import public_api
+from great_expectations.checkpoint.checkpoint import Checkpoint
+from great_expectations.compatibility.typing_extensions import override
+from great_expectations.core.factory.factory import Factory
+from great_expectations.exceptions import DataContextError
+
+if TYPE_CHECKING:
+    from great_expectations.data_context.data_context.abstract_data_context import (
+        AbstractDataContext,
+    )
+    from great_expectations.data_context.store import CheckpointStore
+
+
+# TODO: Add analytics as needed
+class CheckpointFactory(Factory[Checkpoint]):
+    def __init__(self, store: CheckpointStore, context: AbstractDataContext):
+        self._store = store
+        self._context = context
```

Comment:
Checkpoints currently have a reliance on contexts - I'd really like to decouple this but for the time being, the factory needs a pointer here so it can create checkpoints